graphql_compiler/compiler/emit_sql.py

```diff
@@ -144,10 +226,11 @@ class SQLFoldObject(object):
     # ON ...
     # GROUP BY OuterVertex.SOME_COLUMN
 
-    def __init__(self, outer_vertex_table, join_descriptor):
+    def __init__(self, dialect, outer_vertex_table, join_descriptor):
         """Create an SQLFoldObject with table, type, and join information supplied by the IR.
 
         Args:
+            dialect: string passed in from the schema indicating the target dialect
```

Comment:
Are we sure this should be a string and not a SQLAlchemy dialect object? What are the pros/cons of each choice?

Reply:
Yeah so we could imagine changing `dialect` to the actual compiler used, then doing some sort of check like `dialect == sqlalchemy.dialects.mssql.dialect()`, but for some reason different calls to dialect() aren't equal to one another :( 

We could also try something like `isinstance(dialect,MSDialect_pyodbc)`. A pro of this is its immutable, its a constant provided by SQLAlchemy, and it is not liable to change much between different releases of SQLAlchemy.

The pros of the string way of doing things is that it is readable, and that we are going to need to use some constant field of the SQLALchemy compiler object since different instances of the same compiler dialect aren't equal. A downside might be that strings are mutable, and SQLAlchemy might change how they represent different dialects some day so that they no longer use string names, but some other identifier. 

I will change this to use the type check instead of passing strings around. Let me know if you object